src/dialogs/PreviewUploadOverlay.cc

```diff
@@ -0,0 +1,181 @@
+/*
+ * nheko Copyright (C) 2017  Konstantinos Sideris <siderisk@auth.gr>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <QApplication>
+#include <QBuffer>
+#include <QDebug>
+#include <QFile>
+#include <QFileInfo>
+#include <QHBoxLayout>
+#include <QMimeDatabase>
+#include <QVBoxLayout>
+
+#include "Config.h"
+#include "Utils.h"
+
+#include "dialogs/PreviewUploadOverlay.h"
+
+using namespace dialogs;
+
+static constexpr const char *DEFAULT = "Upload %1?";
+static constexpr const char *ERROR   = "Failed to load image type '%1'. Continue upload?";
+
+PreviewUploadOverlay::PreviewUploadOverlay(QWidget *parent)
+  : QWidget{parent}
+  , titleLabel_{this}
+  , fileName_{this}
+  , upload_{tr("Upload"), this}
+  , cancel_{tr("Cancel"), this}
+{
+        auto hlayout = new QHBoxLayout;
+        hlayout->addWidget(&upload_);
+        hlayout->addWidget(&cancel_);
+
+        auto vlayout = new QVBoxLayout{this};
+        vlayout->addWidget(&titleLabel_);
+        vlayout->addWidget(&infoLabel_);
+        vlayout->addWidget(&fileName_);
+        vlayout->addLayout(hlayout);
+
+        connect(&upload_, &QPushButton::clicked, [&]() {
+                emit confirmUpload(data_, mediaType_, fileName_.text());
+                close();
+        });
+        connect(&cancel_, &QPushButton::clicked, [&]() { close(); });
+}
+
+void
+PreviewUploadOverlay::init()
+{
+        auto window   = QApplication::activeWindow();
+        auto winsize  = window->frameGeometry().size();
+        auto center   = window->frameGeometry().center();
+        auto img_size = image_.size();
+
+        fileName_.setText(QFileInfo{filePath_}.fileName());
```

Comment:
`filePath_` is not guaranteed to refer to a real file. Do we have a guarantee that `QFileInfo` will remain well-behaved in that case?

Reply:
Yes it is guaranteed. [`QFileInfo` does not require the file to exist](http://doc.qt.io/qt-5.10/qfileinfo.html#exists).